lib/archethic/utils.ex

```diff
@@ -977,6 +977,30 @@ defmodule Archethic.Utils do
 
   def get_token_properties(_, _), do: {:error, :not_a_token_transaction}
 
+  @doc """
+  Run given function in a task and ensure that it is run at most once concurrently
+  """
+  def run_at_most_once_concurrently(fun, key) do
+    registry = Archethic.AtMostOnceConcurrentlyRegistry
+
+    Task.Supervisor.async_nolink(
+      Archethic.TaskSupervisor,
+      fn ->
+        case Registry.lookup(registry, key) do
```

Comment:
I think you can make the lookup before the process's spawning, otherwise the process would be created for nothing.